compiler/glue-compiler.cc

```diff
@@ -242,6 +242,35 @@ static hilti::rt::Port extract_port(const std::string& chunk, size_t* i) {
     return {static_cast<uint16_t>(port), proto};
 }
 
+static std::vector<hilti::rt::Port> extract_ports(const std::string& chunk, size_t* i) {
+    auto start = extract_port(chunk, i);
+    auto end = std::optional<hilti::rt::Port>();
+
+    if ( looking_at(chunk, *i, "-") ) {
+        eat_token(chunk, i, "-");
+        end = extract_port(chunk, i);
+    }
+
+    if ( end ) {
+        if ( start.protocol() != end->protocol() )
+            throw ParseError("start and end of port range must have same protocol");
+
+        if ( start.port() > end->port() )
+            throw ParseError("start of port range cannot be after its end");
+    }
+
+    std::vector<hilti::rt::Port> result;
+
+    // Port ranges are a closed interval.
+    for ( auto port = start.port(); ! end || port <= end->port(); ++port ) {
+        result.emplace_back(port, start.protocol());
+        if ( ! end )
+            break;
+    }
```

Comment:
Not following the termination condition here: is `! end || port <= end->port()` correct? And we have an `if` terminating in the `!end`, so won't get into 2nd round with that anyways.

Reply:
I wanted to reuse the same loop to handle both the case of `end` being unset (single port) and being set (port range given). Unless I have made a mistake, the logic should be:

-  for `! end` we should make exactly one iteration of the loop which adds `start` to the result,
- else `end` will be set and we want to iterate until and including when `port == end.port()` (closed range).

The alternative here would be to place `start` in `result` before the loop, and then iterate from `start.port() + 1`. This would need additional handling for the case where `start.port() == 65535`, so I went for this logic.

If you think it would help to make this more explicit I can expand the code and/or add additional comments.

Reply:
Ok, got it now.